gdx/src/com/badlogic/gdx/utils/Pools.java

```diff
@@ -16,104 +16,126 @@
 
 package com.badlogic.gdx.utils;
 
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Net.HttpRequest;
+import com.badlogic.gdx.graphics.g2d.GlyphLayout;
+import com.badlogic.gdx.graphics.g2d.GlyphLayout.GlyphRun;
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.Stage.TouchFocus;
+import com.badlogic.gdx.scenes.scene2d.actions.*;
+import com.badlogic.gdx.scenes.scene2d.ui.Table.DebugRect;
+import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener.ChangeEvent;
+import com.badlogic.gdx.scenes.scene2d.utils.FocusListener.FocusEvent;
 import com.badlogic.gdx.utils.DefaultPool.PoolSupplier;
 
 /** Stores a map of {@link Pool}s by type for convenient static access.
  * @author Nathan Sweet */
 public class Pools {
 	static private final ObjectMap<Class<?>, Pool<?>> typePools = new ObjectMap<>();
-	static private final ObjectMap<Class<?>, Pool<?>> supplierPoolsCache = new ObjectMap<>();
+	static public boolean WARN_ON_REFLECTION_POOL_CREATION = true;
+	static public boolean THROW_ON_REFLECTION_POOL_CREATION = false;
+
+	static {
```

Comment:
I uh... don't like this at all. It breaks any kind of separation of concerns, since any change in libGDX that uses Pools needs to also change the source of Pools. Why on Earth can't we create a DefaultPool in the file where one is used? If we did that, we could also choose whether or not for it to be static, in case someone potentially doesn't want threading to break due to a static pool. I don't know if we have any benchmarks for recent Android (post-Dalvik, since our MinSDK now means we won't ever run on a Dalvik-using OS) that can verify if pooling has a performance benefit, or if it's just breaking threading for no current reason. I don't have any clue how people benchmark on Android, either.

Reply:
As I said, we could do this in the individual files, like the `InputEvent` example in the original topic.

Unfortunately, we can't really create a `DefaultPool` in all of the cases. There's at least one api which accepts a `<T extends Action>` which we can't hold the unlimited potential `Pool`s locally. 

https://github.com/libgdx/libgdx/blob/c9638eabb78290bdfcd18b38570f4f91343f8a99/gdx/src/com/badlogic/gdx/scenes/scene2d/actions/Actions.java#L46


I personally, do manage all of the Pools in my own codebase myself, but the `Pools` api is public. If we want this continue to work, I don't see a better option compatible with old Android and no-reflection usage.

Reply:
Following these premises, shouldn't we at least replace internal usages to preferred `DefaultPool` where possible leaving `Actions` to be the exception?

Reply:
Yes, i think those are probably out of scope for the reason of the PR